app/models/application_reference.rb

```diff
@@ -14,6 +14,8 @@ class ApplicationReference < ApplicationRecord
 
   audited associated_with: :application_form
 
+  serialize :questionnaire
```

Comment:
I don't think this is needed for JSONB columns in Rails these days. In fact it seems to cause activerecord to serialize YAML into the JSONB column (as a string) so it messes up the data structure.

With:

```
├─[ RECORD 3 ]────────────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ id                      │ 239                                                                                                                                         │
│ email_address           │ bob@example.com                                                                                                                             │
│ feedback                │ good person                                                                                                                                 │
│ application_form_id     │ 118                                                                                                                                         │
│ created_at              │ 2020-02-06 11:32:46.586137                                                                                                                  │
│ updated_at              │ 2020-02-06 15:46:43.887547                                                                                                                  │
│ name                    │ Bob                                                                                                                                         │
│ relationship            │ good mate                                                                                                                                   │
│ hashed_sign_in_token    │ f89f9d20d3c58a203a5ada74e199868b44cdcca47321007850674766b2459e49                                                                            │
│ consent_to_be_contacted │ t                                                                                                                                           │
│ feedback_status         │ feedback_provided                                                                                                                           │
│ questionnaire           │ "---\nexperience_rating: poor\nexperience_explanation: I was eating my dinner\nguidance_rating: ok\nguidance_explanation: Not much to say r…│
│                         │…eally\nsafe_to_work_with_children: 'false'\nsafe_to_work_with_children_explanation: Because he's a bit eccentric\nconsent_to_be_contacted: …│
│                         │…'true'\nconsent_to_be_contacted_details: feel free 012345678\n"                                                                             │

```

Without:
```
├─[ RECORD 4 ]────────────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ id                      │ 238                                                                                                                                         │
│ email_address           │ billy@example.com                                                                                                                           │
│ feedback                │ yeah alright                                                                                                                                │
│ application_form_id     │ 118                                                                                                                                         │
│ created_at              │ 2020-02-06 11:32:36.54928                                                                                                                   │
│ updated_at              │ 2020-02-06 15:55:53.339515                                                                                                                  │
│ name                    │ Billy                                                                                                                                       │
│ relationship            │ mate                                                                                                                                        │
│ hashed_sign_in_token    │ f3cfb4808e3ee07d640047165106ebe70aaeb3f517f1ff33601c4368e1648d89                                                                            │
│ consent_to_be_contacted │ t                                                                                                                                           │
│ feedback_status         │ feedback_provided                                                                                                                           │
│ questionnaire           │ {"guidance_rating": "very_poor", "experience_rating": "poor", "guidance_explanation": "Didn't have me glasses on so couldn't read it", "exp…│
│                         │…erience_explanation": "Was watching telly", "consent_to_be_contacted": "true", "safe_to_work_with_children": "true", "consent_to_be_contact…│
│                         │…ed_details": "go on then 0898 1234123", "safe_to_work_with_children_explanation": null}                                                     │

```

Reply:
Thanks for this! I had no idea.

Reply:
Nor did I really until I saw some funny looking JSON in the database!